cupy/cuda/cupy_thrust.cu

```diff
@@ -349,7 +349,17 @@ struct _sort {
             transform(cuda::par(alloc).on(stream_),
                       make_counting_iterator<size_t>(0),
                       make_counting_iterator<size_t>(size),
+                      #ifdef __HIP_PLATFORM_HCC__
```

Comment:
@AdrianAbeyta , once upstream PR is done, make sure to check this in the next merge